records_mover/db/mysql/load_options.py

```diff
@@ -126,7 +113,7 @@ def generate_load_data_sql(self,
 def mysql_load_options(unhandled_hints: Set[str],
                        records_format: DelimitedRecordsFormat,
                        fail_if_cant_handle_hint: bool) -> MySqlLoadOptions:
-    hints = records_format.hints
+    hints = records_format.validate(fail_if_cant_handle_hint=fail_if_cant_handle_hint)
```

Comment:
Here's the key call, which based on the value of `fail_if_cant_handle_hint`, either raises an exception or converts hint values into their default values if they are not recognized.